.github/workflows/pre-release.yml

```diff
@@ -11,11 +15,70 @@ defaults:
   run:
     shell: bash
 jobs:
+  prepare:
+    name: Prepare release run
+    runs-on: ubuntu-20.04
+    outputs:
+      next-version: ${{ steps.version_number.outputs.next-version }}
+      branch: ${{ steps.current_branch.outputs.branch }}
+    steps:
+      - name: Check SemVer input
+        env:
+          SEMVER_TYPE: ${{ github.event.inputs.semver-type }}
+        run: |
+          if [[ ! -z "$SEMVER_TYPE" ]]; then
+            echo "SemVer Type is defined. Checking for valid SemVer type..."
+            if [[ "$SEMVER_TYPE" == "major" ]] || [[ "$SEMVER_TYPE" == "minor" ]] || [[ "$SEMVER_TYPE" == "patch" ]]; then
+              echo "::notice::SemVer Type is correctly set to $SEMVER_TYPE! Continuing with this version bump..."
+            else
+              echo "::error::ERROR: Enforced SemVer does not match any of [major,minor,patch]!"
+              echo "Exiting..."
+              exit 1
+            fi
+          else
+            echo "::notice::No SemVer type defined, continuing with auto generated version number..."
+          fi
+
+      - name: Checkout repo
+        uses: actions/checkout@v2.4.0
+        with:
+          fetch-depth: 0
+
+      - name: Find next version number
+        id: version_number
+        env:
+          SEMVER_TYPE: ${{ github.event.inputs.semver-type }}
+        run: |
+          if [[ ! -z "$SEMVER_TYPE" ]]; then
+            npx standard-version@^9.3.1 \
+              --skip.commit \
+              --skip.tag \
+              --skip.changelog \
+              --prerelease "${{ env.PRERELEASE_KEYWORD }}" \
+              --release-as "$SEMVER_TYPE"
+          else
+            npx standard-version@^9.3.1 \
+              --skip.commit \
+              --skip.tag \
+              --skip.changelog \
+              --prerelease "${{ env.PRERELEASE_KEYWORD }}"
+          fi
+
+          NEXT_VERSION=$(cat VERSION.txt)
+          echo "::set-output name=next-version::${NEXT_VERSION}"
+          git checkout HEAD -- VERSION.txt
+
+      - name: Find current branch
+        id: current_branch
+        run: |
+          branch=${GITHUB_REF#refs/heads/}
+          echo "::set-output name=branch::${branch}"
   test:
+    needs: prepare
     strategy:
       matrix:
-        go-version: [1.16.x]
-        platform: [ubuntu-latest]
+        go-version: [ 1.16.x ]
```

Comment:
should we maybe switch to go 1.17 already here, since we plan to upgrade all services in the near future?

Reply:
I would like to update the go-version in one go (hehe) and not gradually. Otherwise we could have build errors due to conflicting versions and such..

Reply:
Also if we test with 1.17 and build for release with 1.16 we could get results that are not well reproducible

Reply:
that makes sense, thank you :)